modules/openxOrtbBidAdapter.js

```diff
@@ -24,6 +22,99 @@ export const spec = {
 
 registerBidder(spec);
 
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    mergeDeep(imp, {
+      tagid: bidRequest.params.unit,
+      ext: {
+        divid: bidRequest.adUnitCode
+      }
+    });
+    if (bidRequest.params.customParams) {
+      utils.deepSetValue(imp, 'ext.customParams', bidRequest.params.customParams);
+    }
+    if (bidRequest.params.customFloor && !imp.bidfloor) {
+      imp.bidfloor = bidRequest.params.customFloor;
+    }
+    return imp;
+  },
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+      ext: {
+        bc: `${bidderConfig}_${bidderVersion}`
+      },
+      regs: {
+        // always set regs.coppa, even when not configured
+        // TODO: is this necessary?
+        coppa: config.getConfig('coppa') ? 1 : 0
+      }
+    })
+    const bid = context.bidRequests[0];
+    if (bid.params.doNotTrack) {
+      utils.deepSetValue(req, 'device.dnt', 1);
+    }
+    if (bid.params.platform) {
+      utils.deepSetValue(req, 'ext.platform', bid.params.platform);
+    }
+    if (bid.params.delDomain) {
+      utils.deepSetValue(req, 'ext.delDomain', bid.params.delDomain);
+    }
+    if (bid.params.test) {
+      req.test = 1
+    }
+    return req;
+  },
+  bidResponse(buildBidResponse, bid, context) {
+    const bidResponse = buildBidResponse(bid, context);
+    if (bid.ext) {
+      bidResponse.meta.networkId = bid.ext.dsp_id;
+      bidResponse.meta.advertiserId = bid.ext.buyer_id;
+      bidResponse.meta.brandId = bid.ext.brand_id;
+    }
+    const {ortbResponse} = context;
+    if (ortbResponse.ext && ortbResponse.ext.paf) {
+      // TODO: the following is likely incorrect when an ORTB response contains more than one bid
+      // all of them would be set with the same content_id (from the last bid), since they share the same `paf` object
```

Comment:
yes thank you you are right the paf object needs to be cloned

Reply:
I'll update it in this PR, but this might take a while to get released so if you need the fix sooner you'll want to open one yourself.